ncli/ncli_db.nim

```
@@ -109,14 +115,46 @@ type
         defaultValue: 1
         desc: "Number of eras to write".}: uint64
 
+    of validatorPerf:
+      perfSlot* {.
+        defaultValue: 0
+        name: "start-slot"
+        desc: "Starting slot, negative = backwards from head".}: int64
+      perfSlots* {.
+        defaultValue: 50000
```

I don't really understand these defaults, but maybe I'm just not imagining the targeted use cases for when specifically getting the validator performance from slots 0 through 50k is of primary interest (but not beyond). Part of this might be the relative slowness of doing all those state transitions, but then the default output is potentially misleading: it doesn't make clear that it's showing only the statistics from the selected subset of slots.

Yes, this is documented in the command-line option help, but that's relatively easy to not have to see to get this subcommand to apparently function, or to overlook the nuance of.

It looks like it's designed as a machine-readable csv, in which case I'm not sure what a better alternative is given these apparent constraints, though.

---

yeah, they're not great - maybe it would make more sense to use the last 100 epochs.

csv is used to make it easier to massage the data afterwards

---

That'd approximately line up with what people expect from beaconcha.in, from what I can tell.